cmd/jaeger/internal/extension/jaegerquery/server.go

```diff
@@ -28,16 +28,19 @@ var (
 )
 
 type server struct {
-	config  *Config
-	logger  *zap.Logger
-	server  *queryApp.Server
-	jtracer *jtracer.JTracer
+	config *Config
+	logger *zap.Logger
+	server *queryApp.Server
+	tracer trace.TracerProvider
+	hcFunc func(*component.StatusEvent)
```

Comment:
there is no need to capture these as individual fields, just create telemetry.Setting in newServer() and store a single telset field